### PR TITLE
Ijaegle nonlinear

### PR DESCRIPTION
**Pull request by igjaegle**
Add:
1/ version 3 of non-linear correction
2/ Updates eta to 2g and 6g skimmer

### CODE REVIEW

**Comment by gluex** [comment]
Test status for this pull request: FAILURE
 
 Failures: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/tests/failures.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/tests-failures.txt)
 Summary: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/tests/summary.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/tests-summary.txt)
 Logs: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/tests/log](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/tests-logs)
 
 Build log: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/make_ijaegle-nonlinear.log](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/make_ijaegle-nonlinear.log)
 Build report: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/report_ijaegle-nonlinear.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/report_ijaegle-nonlinear.txt)
 Location of build: /work/halld/pull_request_test/halld_recon^ijaegle-nonlinear

**Comment by gluex** [comment]
Build status for this pull request: FAILURE
 
 Build log: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/make_ijaegle-nonlinear.log](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/make_ijaegle-nonlinear.log)
 Build report: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/report_ijaegle-nonlinear.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/report_ijaegle-nonlinear.txt)
 Location of build: /work/halld/pull_request_test/halld_recon^ijaegle-nonlinear

**Comment by gluex** [comment]
Test status for this pull request: FAILURE
 
 Failures: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/tests/failures.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/tests-failures.txt)
 Summary: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/tests/summary.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/tests-summary.txt)
 Logs: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/tests/log](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/tests-logs)
 
 Build log: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/make_ijaegle-nonlinear.log](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/make_ijaegle-nonlinear.log)
 Build report: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/report_ijaegle-nonlinear.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/report_ijaegle-nonlinear.txt)
 Location of build: /work/halld/pull_request_test/halld_recon^ijaegle-nonlinear

**Comment by gluex** [comment]
Test status for this pull request: SUCCESS
 
  Summary: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/tests/summary.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/tests-summary.txt)
 Logs: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/tests/log](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/tests-logs)
 
 Build log: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/make_ijaegle-nonlinear.log](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/make_ijaegle-nonlinear.log)
 Build report: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/report_ijaegle-nonlinear.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/report_ijaegle-nonlinear.txt)
 Location of build: /work/halld/pull_request_test/halld_recon^ijaegle-nonlinear

File: src/libraries/FCAL/DFCALShower_factory.cc [hunk 1]
```diff
@@ -203,6 +207,21 @@ jerror_t DFCALShower_factory::brun(JEventLoop *loop, int32_t runnumber)
 	}
       }
     }
+    loop->GetCalib("FCAL/nonlinear", nonlinear);
```
Comment (sdobbs) [inline src/libraries/FCAL/DFCALShower_factory.cc:210]: I am not crazy about the name of this table.  Can it be more descriptive, like "FCAL/nonlinear_correction", or something like that?

**Comment by igjaegle** [comment]
Sure, no problem, I will change to FCAL/nonlinear_correction

tks ig.

**Comment by sdobbs** [comment]
I think this generally looks fine, but since we're working with this section of the code, can we add a CCDB table that specifies which non-linear correction to use?  It can just contain a single integer, which would select which nonlinear correction to use.

**Comment by sdobbs** [comment]
I'm not feeling very creative with names this morning, maybe "FCAL/nonlinear_correction_form"?
Other suggestions are welcome.

**Comment by gluex** [comment]
Build status for this pull request: FAILURE
 
 Build log: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/make_ijaegle-nonlinear.log](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/make_ijaegle-nonlinear.log)
 Build report: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/report_ijaegle-nonlinear.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/report_ijaegle-nonlinear.txt)
 Location of build: /work/halld/pull_request_test/halld_recon^ijaegle-nonlinear

**Comment by gluex** [comment]
Test status for this pull request: SUCCESS
 
  Summary: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/tests/summary.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/tests-summary.txt)
 Logs: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/tests/log](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/tests-logs)
 
 Build log: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/make_ijaegle-nonlinear.log](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/make_ijaegle-nonlinear.log)
 Build report: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/report_ijaegle-nonlinear.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/report_ijaegle-nonlinear.txt)
 Location of build: /work/halld/pull_request_test/halld_recon^ijaegle-nonlinear

**Comment by igjaegle** [comment]
There are some inconsistencies that I am correcting. I should push a new version soon.

**Comment by gluex** [comment]
Test status for this pull request: SUCCESS
 
  Summary: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/tests/summary.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/tests-summary.txt)
 Logs: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/tests/log](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/tests-logs)
 
 Build log: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/make_ijaegle-nonlinear.log](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/make_ijaegle-nonlinear.log)
 Build report: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/report_ijaegle-nonlinear.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/report_ijaegle-nonlinear.txt)
 Location of build: /work/halld/pull_request_test/halld_recon^ijaegle-nonlinear

**Comment by sdobbs** [comment]
Thanks a lot!  looks good except for two things:
* I think that you accidentally put in a file `src/libraries/DFCALShower_factory.cc` (outside of libraries/FCAL)
* Can you also add in values for the default variation for the other run periods for /FCAL/nonlinear_correction_type?  It looks like these values are mostly in a separate variation.

**Comment by igjaegle** [comment]
Thank you, Sean. 
1/ I removed the misplaced file and check that the correct file in libraries/FCAL
2/ I added to DB the /FCAL/nonlinear_correction_type for all calibrated run periods.

**Comment by gluex** [comment]
Test status for this pull request: SUCCESS
 
  Summary: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/tests/summary.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/tests-summary.txt)
 Logs: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/tests/log](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/tests-logs)
 
 Build log: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/make_ijaegle-nonlinear.log](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/make_ijaegle-nonlinear.log)
 Build report: [/work/halld/pull_request_test/halld_recon^ijaegle-nonlinear/report_ijaegle-nonlinear.txt](https://halldweb.jlab.org/pull_request_test/halld_recon^ijaegle-nonlinear/report_ijaegle-nonlinear.txt)
 Location of build: /work/halld/pull_request_test/halld_recon^ijaegle-nonlinear

**Review by sdobbs — APPROVED** [review]
looks good, thanks!

**Pull request merged**